js/query.js

```diff
@@ -1,70 +1,452 @@
+
+                        var basics = {
+                            'gene-symbol': topHit.symbol,
+                            'gene-name': topHit.name,
+                            'gene-id': {
+                                db: 'https://www.ncbi.nlm.nih.gov/gene/',
+                                ident: topHit._id
+                            },
+                            'match-score': topHit._score,
+                            'hits': data.hits.length
+                        };
+
+                        displayData(basics);
+                        displayHeadings();
+                        annotateGene(topHit._id);
+                    } else {
+                        var empty = {
+                            hits: 'No hits',
+                            matchScore: 0
+                        };
+
+                        displayData(empty);
+                        hideData(document.getElementById('info-div'));
+                        hideData(document.getElementById('loc-div'));
+                        hideData(document.getElementById('summary-div'));
+                        hideData(document.getElementById('species-div'));
+                        hideHeadings();
                     }
-                )
-                .catch(function(err) {
-                    console.error('Fetch Query Error', err);
                 });
+            })
+            .catch(function(err) {
+                console.error('Fetch Query Error', err);
+            });
+    }
+
+    function displayData(dataObj) {
+        for (data in dataObj) {
+            if (typeof dataObj[data] !== 'undefined' && dataObj[data] !== null) {
+                currentData = document.getElementById(data);
+                currentData.classList.remove('hidden');
+                currentLabel = document.getElementById(data + '-label');
+                currentLabel.classList.remove('hidden');
+
+                // Add new/remove old links from appropriate divs.
+                if (currentData.classList.contains('addlink')) {
+                    var oldLinks = currentData.getElementsByTagName('a');
+                    while (oldLinks.length > 0) {
+                        oldLinks[0].parentNode.removeChild(oldLinks[0]);
+                    }
+                    var linkData = dataObj[data];
+                    var link = linkData['db'] + linkData['ident'];
+                    var aTag = document.createElement('a');
+                    aTag.setAttribute('href', link);
+                    aTag.textContent = linkData['ident'];
+                    currentData.appendChild(aTag);
+                } else {
+                    currentData.textContent = dataObj[data];
+                }
+            } else {
+                // Hide and clear any previous results.
+                currentData = document.getElementById(data);
+                if (!currentData.classList.contains('hidden')) {
+                    currentData.textContent = '';
```

Comment:
Is this call necessary? As far as I can tell, the next time this element's `display` is set to visible, its text will already have been replaced with the latest search results, so there's no need to erase the text.

I just tried testing the page with this and the other similar call commented out and as far as I can tell I don't notice a difference.

Reply:
Correct and removed.